examples/hubert/lightning.py

```diff
@@ -0,0 +1,137 @@
+from typing import Tuple
+
+import torch
+import torchaudio
+from dataset import BucketizeSampler, DistributedBatchSampler, HuBERTDataSet, CollateFnHubert
+from loss import hubert_loss
+from pytorch_lightning import LightningModule
+from torch import Tensor
+from torch.optim.optimizer import Optimizer
+from torch.utils.data import DataLoader
+
+
+Batch = Tuple[Tensor, Tensor, Tensor]
+
+
+class LinearDecayLRScheduler(torch.optim.lr_scheduler._LRScheduler):
+    """Linear learning rate scheduler with warm up."""
+
+    def __init__(
+        self,
+        optimizer: Optimizer,
+        warmup_updates: int,
+        max_updates: int,
+        last_epoch: int = -1,
+        verbose: bool = False,
+    ):
+        self.warmup_updates = warmup_updates
+        self.max_updates = max_updates
+        super().__init__(optimizer, last_epoch=last_epoch, verbose=verbose)
+
+    def get_lr(self):
+        if self._step_count <= self.warmup_updates:
+            return [self._step_count / self.warmup_updates * base_lr for base_lr in self.base_lrs]
+        elif self.optimizer._step_count >= self.max_updates:
```

Comment:
might want to check on whether `self._step_count` and `self.optimizer._step_count` differ in any way, and then stick to using just one throughout this function

Reply:
I tested the `_step_count` of optimizer and lr_scheduler in [this script](https://gist.github.com/nateanl/6dbae8b49320f87b1452680f01a044f5). `self._step_count` makes more sense because the optimizer's step_count starts with 0. If we use `optimizer._step_count`, the first learning rate will be 0.